### ISSUE TITLE
appLayout,How can I control page-content height,after appLayout excuted.

### ISSUE THREAD

**Issue by zhangwen9229** [kikinteractive/app#112]
I want to make a footer. So I shoul change the app-content's height.Now,I change it at appShow time.
 But,when focus input,viewport height changed. app-content be restored.

Please help me resolve it.

**Comment by muten84** [comment]
Hi, I have a fork of this project in my repo that do exactly what you want defining a CSS class app-bottombar.

**Comment by zhangwen9229** [comment]
Thank you for your reply. I'll go to check it,and try it. later.

**Comment by muten84** [comment]
You are right because you have to define your own. So specify the height props in the .app-bottombar style class in a your css file and it should fix the content height automatically.

**Comment by zhangwen9229** [comment]
@muten84 You had changed app.js? ---- function fixContentHeight()   ???

**Comment by muten84** [comment]
Yes :)

**Comment by zhangwen9229** [comment]
@muten84 Let me try it.  Use app.js under the dist. OK？     ^_^

**Comment by muten84** [comment]
I understand you can try invoke App._layout() it should trigger a content fix.

**Comment by zhangwen9229** [comment]
@muten84 I'm sorry,I find you didn't package it？

**Comment by zhangwen9229** [comment]
@muten84 The contentfix be fired when input focus,and blur. Result,my footer be covered.
App.controller('home', function (page) {
        // put stuff here
      });

      App.controller('page2', function (page) {
        // put stuff here
        $(page).on("appLayout",function(){
           console.log("appLayout");
           $(page).find('.app-content').height(300).css({
             background: 'red'
           });
        }).on("appReady",function(){
            $(page).find('.app-content').height(300).css({
             background: 'red'
           });
        })
      });

      try {
        App.restore();
      } catch (err) {
        App.load('home');
      }

**Comment by muten84** [comment]
I think a rebuild is needed ... try the App._layout() if not working I will commit a built package.

**Comment by zhangwen9229** [comment]
Could you rebuild one.I don't know how to rebuilt it.    (⊙﹏⊙)  My Email： 378253487@qq.com

**Comment by muten84** [comment]
Ok I will commit when possible

**Comment by zhangwen9229** [comment]
@muten84  Thank you. ^_^

**Comment by muten84** [comment]
@zhangwen9229 I have just done the commit of the rebuild https://github.com/muten84/app/commit/87f35728ce2f787caf81766d130941b232aa4de6
You can try using  the .app-bottombar class and the content height should be fixed automatically.

**Comment by zhangwen9229** [comment]
@muten84 Ok,I will try it at night.Then give you a reply. Here is 10:08 am.  ^_^

**Comment by zhangwen9229** [comment]
@muten84 I had try app v3.0.6,yours. but I don't know why you plus 10,here 

content.style.height = (height - (topbarHeight + bottombarHeight +10 )) + 'px'

line 1759. 
please check it.

**Comment by muten84** [comment]
You are right this is a +10 margin, forced by the framework. However it's not mandatory, and you can avoid this using margin-top property with a negative value. 
I don't understand the overflow-x property was set to...

**Comment by zhangwen9229** [comment]
@muten84 overflow-x , mean:  Horizontal scroll bar,appear in app-content .

**Comment by zhangwen9229** [comment]
@muten84 I tested app-bottombar,it worked;But,when I add app-bottombar to next page , the previous page's event,appReady will not be triggered;
